services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/email/FilteredEmailSenderServiceITest.kt

```diff
@@ -46,19 +46,42 @@ class EmailSenderServicePropertiesITest : DatabaseTest() {
     }
 
     @Test
-    fun `sendJohtoselvitysCompleteEmail when recipient not in allow list does not send`() {
+    fun `sendJohtoselvitysCompleteEmail blocks send when recipient is not in allow list`() {
         emailSenderService.sendJohtoselvitysCompleteEmail("foo@bar.test", 13L, "JS2300001")
 
         assertThat(greenMail.receivedMessages.size).isEqualTo(0)
     }
 
     @Test
-    fun `sendApplicationNotificationEmail when user management not enabled does not send`() {
+    fun `sendJohtoselvitysCompleteEmail blocks send when recipient is close to an allowed email`() {
+        emailSenderService.sendJohtoselvitysCompleteEmail("atest@test.test", 13L, "JS2300001")
+
+        assertThat(greenMail.receivedMessages.size).isEqualTo(0)
+    }
+
+    @Test
+    fun `sendJohtoselvitysCompleteEmail blocks send when dot is replaced`() {
+        emailSenderService.sendJohtoselvitysCompleteEmail("test@test-test", 13L, "JS2300001")
+
+        assertThat(greenMail.receivedMessages.size).isEqualTo(0)
+    }
+
+    @Test
+    fun `sendJohtoselvitysCompleteEmail sends email when email matches wildcard`() {
```

Comment:
Should there also be a test for email addresses like `test+suffix@wildcard.com`?

Reply:
I added a test.